### PR TITLE
Add prettier

### PR DESCRIPTION
**Pull request by weeman1337**
https://github.com/matrix-org/eslint-plugin-matrix-org/pull/35 to be merged before
 
Should best be reviewed commit wise

## Checklist

* [x] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: spec/integ/megolm-integ.spec.ts [hunk 1]
```diff
@@ -1010,70 +1010,73 @@ describe("megolm", () => {
...
-        });
-
-        // encrypt a message with the group session
-        const messageEncrypted = encryptMegolmEvent({
-            senderKey: testSenderKey,
-            groupSession: groupSession,
-            room_id: ROOM_ID,
-        });
+    it(
+        "should successfully decrypt bundled redaction events " + "that don't include a room_id in their /sync data",
```
Comment (weeman1337) [inline spec/integ/megolm-integ.spec.ts:1014]: Looks silly, but if you remove the concat, prettier puts `it`, the string and `async () => {` into a single line. Then ESLint starts complaining about hitting the hard line length limit. :man_shrugging:

Comment (t3chguy) [reply spec/integ/megolm-integ.spec.ts:1014]: If prettier doesn't comprehend our hard line limit then this will be really painful to have to conjure

Comment (weeman1337) [reply spec/integ/megolm-integ.spec.ts:1014]: They added something about it in their docs
https://prettier.io/docs/en/options.html#print-width

As far as I understand that means use prettier or disable length checks in your other linters.

Comment (t3chguy) [reply spec/integ/megolm-integ.spec.ts:1014]: Sounds like we'll need to ask for people's opinions on this, conveniently it is thursday

**Review by turt2live — COMMENTED** [review]
I haven't looked at this in depth, but believe we should probably land it soon (particularly considering the project is now fully TS --strict compliant, which means a boatload of conflicts are pending)

File: spec/integ/megolm-integ.spec.ts (hunk 1 above)
Comment (weeman1337) [reply spec/integ/megolm-integ.spec.ts]: We decided to disable `max-len` in ESLint

**Review by andybalaam — APPROVED** [review]
Do it!

File: .eslintrc.js [hunk 2]
```diff
@@ -1,13 +1,6 @@
 module.exports = {
```
Comment (MadLittleMods) [inline .eslintrc.js:1]: We should add these bulk fix commits to [`.git-blame-ignore-revs`](https://github.com/matrix-org/matrix-js-sdk/blob/193c38523c6304c86e56d6bdc5089ece26ad34da/.git-blame-ignore-revs) so we can easily skip over them in the git blame interface

(same with other bulk Typescript fixes that have happened recently)

Comment (weeman1337) [reply .eslintrc.js:1]: Yes, definitively. I will open PRs later :+1:

**Pull request merged**